src/middleware.ts

```diff
@@ -0,0 +1,42 @@
+import { NextResponse } from "next/server";
+import type { NextRequest } from "next/server";
+import { get } from "@vercel/edge-config";
+
+export const config = {
+  matcher: [
+    /*
+     * Match all request paths except for the ones starting with:
+     * - api (API routes)
+     * - _next/static (static files)
+     * - _next/image (image optimization files)
+     * - favicon.ico (favicon file)
+     */
+    {
+      source: "/((?!api|_next/static|_next/image|favicon.ico).*)",
+      missing: [
+        { type: "header", key: "next-router-prefetch" },
+        { type: "header", key: "purpose", value: "prefetch" },
+      ],
+    },
+  ],
+};
+
+export async function middleware(req: NextRequest) {
+  let isInMaintenanceMode = false;
+
+  try {
+    isInMaintenanceMode =
+      // Don't show the maintenance page in development
+      process.env.NODE_ENV !== "development" &&
```

Comment:
It looks to me like all Vercel deployments use the same store, so I was wondering about how we keep maintenance mode off for the dev deployment for testing (while leaving prod in maintenance mode). Looks like it's this, yeah? Could you add `NODE_ENV` to .env.sample? I'll add it on Vercel, with values `development` and `production`. Thank ya!

Reply:
Oh, there's no need to add `NODE_ENV` manually anywhere. By default it is always 'development' and Next/React add `NODE_ENV=production` to the build without any manual intervention. It's a common convention and doesn't require any extra work to do it's magic. More info: https://nodejs.org/en/learn/getting-started/nodejs-the-difference-between-development-and-production

Reply:
And of course

> Looks like it's this, yeah?

Yes indeed! That's my thinking, but happy to go for a different solution if you like.

Reply:
Amazing, I didn't know all that! Thanks for the explanation 🧠